src/complex-list-field/complex-list-field.component.ts

```diff
@@ -36,16 +36,66 @@ import {
   ],
   templateUrl: './complex-list-field.component.html'
 })
-export class ComplexListFieldComponent extends AbstractListFieldComponent {
+export class ComplexListFieldComponent extends AbstractListFieldComponent implements OnInit {
 
   @Input() values: Array<Object>;
   @Input() schema: Object;
   @Input() path: string;
 
+  keys: Array<Array<string>>;
+
   @Output() onValuesChange: EventEmitter<Array<Object>> = new EventEmitter<Array<Object>>();
 
   constructor(public emptyValueService: EmptyValueService,
     public appGlobalsService: AppGlobalsService) {
     super();
   }
+
+  ngOnInit() {
+    super.ngOnInit();
+
+    this.keys = this.values
+      .map(value => Object.keys(value));
+  }
+
+  onFieldAdd(field: string, index: number) {
+    this.keys[index].push(field);
+  }
+
+  /**
```

Comment:
👍 I think it can make sense, was there a reason not to do it? You think it will be heavy to do the `.map` every time? (might be with big lists of references)